File: leetcode/August_2024/Summary Ranges.cpp
#include <vector>
#include <string>

class Solution
{
public:
    std::vector<std::string> summaryRanges(std::vector<int>& nums)
    {
        std::vector<std::string> result;
        if (nums.empty()) {
            return result;  // Return empty result if input is empty
        }

        int start = nums[0];  // Initialize start of the range
        int prev = nums[0];   // Initialize previous number

        for (int i = 1; i < nums.size(); ++i)
        {
            if (nums[i] == prev + 1) {
                // Continue the range
                prev = nums[i];
            } else {
                // End the current range and start a new one
                if (start == prev) {
                    result.push_back(std::to_string(start));
                } else {
                    result.push_back(std::to_string(start) + "->" + std::to_string(prev));
                }
                start = nums[i];
                prev = nums[i];
            }
        }

        // Add the last range
        if (start == prev) {
            result.push_back(std::to_string(start));
        } else {
            result.push_back(std::to_string(start) + "->" + std::to_string(prev));
        }

        return result;
    }
};

File: leetcode/August_2024/Summary_Ranges.py
class Solution:
    def summaryRanges(self, nums):
        if len(nums) == 0: return []
        start = nums[0]
        last_num = nums[0]
        ans = []
        for num in nums[1:]:
            if num == last_num + 1:
                last_num = num
            else:
                ans.append(f"{start}->{last_num}" if start!= last_num else str(start))
                start = num
                last_num = num
        ans.append(f"{start}->{last_num}" if start!= last_num else str(start))
        return ans

